ttkwidgets/tickscale.py

```diff
@@ -0,0 +1,633 @@
+            elif digits < 0:
+                self._digits = digits
+                self._formatter = '{:g}'
+                reinit = True
+            else:
+                self._digits = digits
+                self._formatter = '{:.' + str(self._digits) + 'f}'
+                interv = self._get_precision(self._tickinterval)
+                resol = self._get_precision(self._resolution)
+                start = kw.get('from', kw.get('from_', self._start))
+                end = kw.get('to', self.scale.cget('to'))
+                from_ = self._get_precision(start)
+                to = self._get_precision(end)
+                d = max(interv, resol, from_, to)
+                if self._digits < d:
+                    self._resolution = float('1e-{}'.format(self._digits))
+                    self._tickinterval = round(self._tickinterval, self._digits)
+                    if self._resolution > self._tickinterval:
+                        self._tickinterval = self._resolution
+                    kw['to'] = round(end, self._digits)
+                    if 'from_' in kw:
+                        del kw['from_']
+                    kw['from'] = round(start, self._digits)
+                reinit = True
+        elif self._digits > 0:
+            start = kw.get('from', kw.get('from_', self._start))
+            end = kw.get('to', self.scale.cget('to'))
+            from_ = self._get_precision(start)
+            to = self._get_precision(end)
+            interv = self._get_precision(self._tickinterval)
+            resol = self._get_precision(self._resolution)
+            digits = max(self._digits, interv, resol, from_, to)
+            if digits != self._digits:
+                self._digits = digits
+                self._formatter = '{:.' + str(self._digits) + 'f}'
+                reinit = True
+        if 'variable' in kw:
+            self._var = kw['variable']
+            if not self._var:
+                self._var = tk.DoubleVar(self, self.get())
+                kw['variable'] = self._var
+            try:
+                self._var.trace_add('write', self._increment)
+            except Exception:
+                # backward compatibility
+                self._var.trace('w', self._increment)
+
+        self.scale.configure(**kw)
+
+        if 'from_' in kw or 'from' in kw or 'to' in kw:
+            self._extent = self.scale.cget('to') - self.scale.cget('from')
+            self._start = self.scale.cget('from')
+            reinit = True
+        if 'style' in kw:
+            self._style_name = kw['style']
+            if not self._style_name:
+                self._style_name = '%s.TScale' % (str(self.scale.cget('orient')).capitalize())
+            self._style_change()
+        if 'orient' in kw:
+            if kw['orient'] == 'vertical':
+                self._style_name = self._style_name.replace('Horizontal', 'Vertical')
+                if 'tickpos' not in kw:
```

Comment:
`tickpos` will never be present in `kw` here, because of line 192: if it was in kw, it is popped from kw and not added back.

Reply:
Corrected. I have put this if statement before popping 'tickpos' from kw.